### PR TITLE
updated image links in mdx files, added some extra features and corrected some as well! search bar implemented

### PR DESCRIPTION
**Pull request by sam-shervin**
- updated all the foreign image links in the mdx files to GitHub image links.
- lookup feature in discord.mdx did not work and so the mdx content code is updated to make it work
- author LinkedIn links made clickable
- highlighted the URLs (made changes in the CSS file)
- corrected the Instagram link in the footer
- Search Bar added
- Changed the header
- added figma prototyping content

### CODE REVIEW

**Review by AdityaSai2004 — APPROVED** [review]
Can proceed with merging

**Review by AdityaSai2004 — APPROVED** [review]
No issues found

**Review by AdityaSai2004 — CHANGES_REQUESTED** [review]
User content can be collected at a later point

**Comment by sam-shervin** [comment]
user content part removed (can be added at a later point)

**Comment by sam-shervin** [comment]
> User content can be collected at a later point

user content part removed (can be added at a later point)

**Review by AdityaSai2004 — APPROVED** [review]
Approved

File: tailwind.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
   darkMode: "class",
-  content: ["./src/**/*.{js,jsx,ts,tsx, md. mdx}"],
```
Comment (AbhijithGanesh) [inline tailwind.config.js:4]: Good catch!

File: src/scripts/scripts.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        document.querySelectorAll("div #con a").forEach((item : any) => item.setAttribute('target', '_blank'))
+    );
+  };
+
+export const content_selector = (document: any) => {
+    return (
+        document.querySelectorAll('div .cledit-section span').forEach((item: any)=> console.log(item.innerText))
+    );
+  };
+
```
Comment (AbhijithGanesh) [inline src/scripts/scripts.tsx:14]: Nice work but I would already add `_blank` Attribute!

File: src/pages/index.tsx [hunk 3]
```diff
@@ -169,7 +171,7 @@ const IndexPage: React.FC<PageProps> = () => {
                 </section>
               </section>
             </section>
-          </section>
+            </section>
```
Comment (AbhijithGanesh) [inline src/pages/index.tsx:174]: Redundant Change

File: src/designs/static.tsx [hunk 4]
```diff
@@ -46,6 +51,9 @@ export default TemplateComponent;
 export const Head = ({ pageContext }: any) => {
   return (
     <>
+      <script>
+      {new_tab(document)}
```
Comment (AbhijithGanesh) [inline src/designs/static.tsx:55]: I'll approve this for the mean time, but this should be done a tad bit better!

File: src/content/working-with-neural-networks.mdx [hunk 5]
```diff
@@ -28,7 +28,7 @@ have dominated in machine intelligence
 - Seaborn([seaborn: statistical data visualization — seaborn 0.12.1 documentation (pydata.org)](https://seaborn.pydata.org/))
 - Tensorflow([https://www.tensorflow.org/](https://www.tensorflow.org/))
 - Keras([https://keras.io/](https://keras.io/))
-- If one has a strong enough pc(with at least 16 GB RAM, M.2 SSD, RTX2080 super , i7 11th gen or ryzen 7 5000 series) they can do it on jupyter([https://jupyter.org/](https://jupyter.org/)) else use keggle([https://www.kaggle.com/](https://www.kaggle.com/))
+- If one has a strong enough pc(with at least 16 GB RAM, M.2 SSD, RTX2080 super , i7 11th gen or ryzen 7 5000 series) they can do it on jupyter([https://jupyter.org/](https://jupyter.org/)) else use kaggle([https://www.kaggle.com/](https://www.kaggle.com/))
```
Comment (AbhijithGanesh) [inline src/content/working-with-neural-networks.mdx:31]: Good Catch

File: src/content/native-android-app.mdx [hunk 6]
```diff
@@ -36,7 +36,7 @@ We are all addicted to our mobile phones and a major reason for that is the vari
 
 ## Roadmap
 
-<img src="https://roadmap.sh/roadmaps/android/roadmap.png" />
+![image](https://user-images.githubusercontent.com/121488288/218552812-a5654850-a1c9-4dd8-9d17-797ff05e019e.png)
```
Comment (AbhijithGanesh) [inline src/content/native-android-app.mdx:39]: Great move!

File: src/content/introduction-to-react-part-one.mdx [hunk 7]
```diff
@@ -14,7 +14,7 @@ React is a free and open-source front-end JavaScript library for building user i
 
 Web development is broadly classified into 2. Front end and Back end. Front end is the interface with which the user interacts/the part of the website visible to the user. While the backend is where all the magic happens as I like to say.
 
-<img src="https://cdn.discordapp.com/attachments/792723261845667851/1055911137968914542/react.png" />
+![image](https://user-images.githubusercontent.com/121488288/218552344-e5aa17cf-5d2c-4785-9a50-f659e027a6a6.png)
```
Comment (AbhijithGanesh) [inline src/content/introduction-to-react-part-one.mdx:17]: Good work again

File: src/content/discord.mdx [hunk 8]
```diff
@@ -308,17 +310,19 @@ For running the bot, save this and you can run it as you would do for a normal f
...
 
 And there, my friends, we end this guide.
 
 ## Wrapping up
 
 We can do a lot of things with Discord bots. The possibilities are endless. To give you a glimpse of the power of Discord bots, here is a picture of a GameBoy Colour emulator running inside of Discord on a Discord bot.
 
-![The image](https://media.discordapp.net/attachments/889873953479356436/1056486630569025586/image.png)
+![image](https://user-images.githubusercontent.com/121488288/218533394-5539bd26-188d-4962-84b8-4a81e8442129.png)
+
```
Comment (AbhijithGanesh) [inline src/content/discord.mdx:325]: Great effotrts @sam-shervin

File: src/content/deep-learning.mdx [hunk 9]
```diff
@@ -218,7 +218,7 @@ This is a video that I found very interesting by 3B1B. Its about the Bayes theor
 
 #### Python
 
-[3.11.1 Documentation](https://docs.python.org/3.11/)
+[Documentation](https://docs.python.org/3.11/)
```
Comment (AbhijithGanesh) [inline src/content/deep-learning.mdx:221]:
```suggestion
[Python Documentation](https://docs.python.org/3.11/)
```

File: src/components/searchbar/searchbar.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
+import React, { useState} from 'react';
+import { graphql, useStaticQuery } from "gatsby";
+import './searchbar.css'
+import { SearchIcon } from "../footer";
+import {  RiCloseFill } from "react-icons/ri";
+
+
+const SearchBar = (placeholder: any): JSX.Element => {
```
Comment (AbhijithGanesh) [inline src/components/searchbar/searchbar.tsx:8]: change the type from `any`

File: src/components/searchbar/searchbar.tsx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+            frontmatter {
+              slug
+              title
+            }
+          }
+        }
+      }
+    `);
+    const [filteredData, setFilteredData] = useState([]);
+    const [wordEntered, setWordEntered] = useState("");
```
Comment (AbhijithGanesh) [inline src/components/searchbar/searchbar.tsx:22]: Type cast according to the state types
```suggestion
    const [filteredData, setFilteredData] = useState<typeof data[]>();
    const [wordEntered, setWordEntered] = useState<string>("");
```

File: package.json [hunk 12]
```diff
@@ -31,6 +31,7 @@
     "gatsby-source-filesystem": "^5.3.1",
     "js-yaml": "^4.1.0",
     "markdown-it": "^13.0.1",
+    "node": "^19.6.0",
```
Comment (AbhijithGanesh) [inline package.json:34]:
```suggestion
    "node": "^19.6.0",
```
Delete this

**Review by AbhijithGanesh — CHANGES_REQUESTED** [review]

File: src/components/searchbar/searchbar.tsx (hunk 11 above)
Comment (sam-shervin) [reply src/components/searchbar/searchbar.tsx]: done

File: package.json (hunk 12 above)
Comment (sam-shervin) [reply package.json]: done

**Comment by sam-shervin** [comment]
> change the type from `any`

I hope most changes have been rectified on my part except for the type of placeholder in searchbar.tsx which I wasn't able to find. I request you review and accept the pull request as soon as possible.

**Pull request closed without merging**